### PR TITLE
Beautify shuffleboard

### PR DESCRIPTION
**Pull request by GoldCrafter**
# Puporse: Created SpikeBoard to organize ShuffleBoard.

## Files Changed:
- build.gradle: updated wpilib
- KeyValue.java & SpikeBoard.java: created SpikeBoard organization system
- RobotContainer, Arm, Drivetrain, & Claw.java: Created ShuffleBoard tabs
- Commands folder: Changed SmartDashboard to SpikeBoard.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Arm.java [hunk 1]
```diff
@@ -215,6 +241,19 @@ private void extendTo(double inches) {
...
+    /**
+     * Returns the position that the arm wants to reach, in encoder units.s
+     * 
+     * @returns double
+     */
+    public double getCommandedEncoderPosition() {
+        return m_pivotCommandedEncoderUnits;
+    }
+
+    public double getCommandedExtentionPosition() {
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/Arm.java:253]: Rename to be clearer: GetCommandedExtenderEncoderPostion

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 2]
```diff
@@ -175,13 +185,17 @@ public void periodic() {
                 Rotation2d.fromDegrees(currentPose.getHeadingDegrees()));
         Logger.getInstance().recordOutput("odometry", currentPose2d);
 
-        // Push robot info to Dashboard
-        SmartDashboard.putNumber("Kinematics X (Feet)", currentPose.getX());
-        SmartDashboard.putNumber("Kinematics Y (Feet)", currentPose.getY());
-        SmartDashboard.putNumber("Kinematics Heading (degrees)", currentPose.getHeadingDegrees());
+        // // Push robot info to Dashboard
+        // SmartDashboard.putNumber("Kinematics X (Feet)", currentPose.getX());
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:189]: If we aren't using this remove the commented out code

**Review by bjmcternan — CHANGES_REQUESTED** [review]
Some things to clean up and fix

File: src/main/java/frc/robot/commands/SequentialAutoCommand.java [hunk 3]
```diff
@@ -60,35 +63,63 @@ public SequentialAutoCommand(Drivetrain drivetrain, Arm arm, Claw claw, Kinemati
...
-				// new Wait(1),
-				// Reset kinematics to the blue left position
-				new DriveToBalance(m_drivetrain),
-				new ResetKinematics(new Position2D(0, 0, Math.toRadians(180)), m_drivetrain, m_kinematics),
-				new DriveBackwards(m_drivetrain, m_kinematics, .23, -3.5),
-				new AutoBalance(m_drivetrain));
+		double firstSpeed = m_autoTab.getDouble("first speed", 0);
+		System.out.println(firstSpeed);
+
+		// addCommands(
```
[the remark this answers is not in the document]
Comment (GoldCrafter) [reply src/main/java/frc/robot/commands/SequentialAutoCommand.java]: Updated to get all speed and distance values from auto tab

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 2 above)
Comment (GoldCrafter) [reply src/main/java/frc/robot/subsystems/Drivetrain.java]: Removed

File: src/main/java/frc/robot/subsystems/Arm.java [hunk 4]
```diff
@@ -215,6 +241,19 @@ private void extendTo(double inches) {
         extenderTalonFX.set(TalonFXControlMode.MotionMagic, encoderUnits, DemandType.ArbitraryFeedForward, 0.0d);
     }
 
+    /**
+     * Returns the position that the arm wants to reach, in encoder units.s
+     * 
+     * @returns double
+     */
+    public double getCommandedEncoderPosition() {
```
[the remark this answers is not in the document]
Comment (GoldCrafter) [reply src/main/java/frc/robot/subsystems/Arm.java]: Renamed

File: src/main/java/frc/robot/subsystems/Arm.java [hunk 5]
```diff
@@ -184,19 +195,33 @@ private void rotateTo(double radians) {
...
         pivotTalonFX.set(TalonFXControlMode.MotionMagic, encoderUnits, DemandType.ArbitraryFeedForward,
                 PIVOT_KF * Math.abs((Math.cos(radians))));
     }
 
+    /**
+     * Get the current encoder units.
+     * 
+     * @return
+     */
+    public double getCurrentEncoderUnits() {
```
[the remark this answers is not in the document]
Comment (GoldCrafter) [reply src/main/java/frc/robot/subsystems/Arm.java]: Renamed

File: src/main/java/frc/robot/classes/SmoothControl.java [hunk 6]
```diff
@@ -62,11 +62,10 @@ public double computeTurnRate(Position2D currentPose, Position2D targetPose, dou
 
   // Equation 14, Calculates the required turn rate
   private double calculateK(double range, double theta, double delta) {
-    double retVal = 0.0d;
-    if(0.0d != m_range) {
-      retVal = -(1 / m_range) * (K2 * (delta - Math.atan(-K1 * theta)) +
+    
+      double retVal = -(1 / range) * (K2 * (delta - Math.atan(-K1 * theta)) +
```
[the remark this answers is not in the document]
Comment (GoldCrafter) [reply src/main/java/frc/robot/classes/SmoothControl.java]: Fixed

**Review by bjmcternan — APPROVED** [review]
Looks good

**Pull request merged**